tests/elevation_service/one_point_per_user.py

```diff
@@ -1,131 +1,164 @@
+                f"?token={ElevationConfig.TOKEN}",
+                json=body,
+                headers={"Content-Type": "application/json"},
+            )
 
+    # def on_stop(self):
+    #     # print(self.body)
+    #     average_response_time = self.environment.runner.stats.total.avg_response_time
+    #     # create_start_time_response_time_graph(graph_name=f"RequestStartTime_vs_ResponseTime-{run_number}",
+    #     #                                       start_time_data=start_time_data, response_time_data=response_time_data)
+    #
+    #     # print(self.request_bodies_cycle)
 
-counters = initiate_counters_by_ranges(config_ranges=percent_ranges)
-total_requests = 0
-run_number = 0
-start_time_data = []
-response_time_data = []
-file_created = False
 
+@events.test_start.add_listener
+def on_locust_init(environment, **_kwargs):
+    environment.users_count = environment.runner.target_user_count
 
-@events.request.add_listener
-def on_request(response_time, **kwargs):
-    start_time = datetime.datetime.fromtimestamp(kwargs['start_time'])
-    response_time_data.append(response_time)
-    start_time_data.append(start_time)
+
+counters = initiate_counters_by_ranges(config_ranges=percent_ranges)
 
 
 @events.request.add_listener
 def response_time_listener(response_time, **kwargs):
     global counters, total_requests
-
-    for index, range_val in enumerate(percent_ranges):
-        if range_val[1] is None and response_time >= range_val[0]:
+    for index, value in enumerate(ranges[:-1]):
+        if value > response_time:
             counters[f"counter{index + 1}"] += 1
-        elif range_val[0] <= response_time <= range_val[1]:
-            counters[f"counter{index + 1}"] += 1
-
+            break
+    if ranges[-2] < response_time:
+        counters[f"counter{len(ranges)}"] += 1
     total_requests += 1
 
+    # if response_time < percent_ranges[0][1]:
+    #     counters[f"counter{0}"] += 1
+    # elif response_time < percent_ranges[1][1]:
+    #     counters[f"counter{1}"] += 1
+    # else:
+    #     counters[f"counter{2}"] += 1
+    # for index, range_rsp in enumerate(percent_ranges):
+    #     if range_rsp[1] is None and response_time >= range_rsp[0]:
+    #         counters[f"counter{index + 1}"] += 1
+    #         break
+    #     elif range_rsp[0] <= response_time <= range_rsp[1]:
+    #         counters[f"counter{index + 1}"] += 1
+    #         break
+    #
+    # total_requests += 1
+
+
+# @events.request.add_listener
+# def on_request(response_time, **kwargs):
+#     global total_requests, start_time_data
+#     start_time = datetime.datetime.fromtimestamp(kwargs['start_time'])
+#     response_time_data.append(response_time)
+#     start_time_data.append(start_time)
+#     total_requests += 1
+
 
 @events.test_start.add_listener
-def reset_counters(**kwargs):
-    global counters, total_requests, run_number, start_time_data, response_time_data
-    counters = counters
+def reset_data_counters(**kwargs):
```

Comment:
General comment : *kwargs is not good practice, better us meaningful names for function param :)